src/main/java/crazypants/enderio/teleport/TravelController.java

```diff
@@ -111,42 +111,55 @@ private TravelController() {
+        // allow 4 blocks of c/s player pos desync
+        if (getPower(toTp, source, target, -4F) > powerUse) return "power use too little";
         ItemStack equippedItem = toTp.getCurrentEquippedItem();
         switch (source) {
             case TELEPAD:
                 // this source is invalid for this type of packet
-                return false;
+                return "invalid travel source";
             case BLOCK:
                 // this source is only triggered when player is on any active travel anchor
                 BlockCoord on = getActiveTravelBlock(toTp);
-                if (on == null) return false;
+                if (on == null) return "on anchor is null";
                 // target must be a valid selectedCoord
                 // selectedCoord can either be a block right above/below when the player is on anchor...
-                if (on.x == x && on.z == z) return true;
+                if (on.x == x && on.z == z) return null;
                 // or another anchor
                 TileEntity maybeAnchor = target.getTileEntity(toTp.worldObj);
-                if (!(maybeAnchor instanceof ITravelAccessable)) return false;
+                if (!(maybeAnchor instanceof ITravelAccessable)) return "not anchor";
                 ITravelAccessable anchor = (ITravelAccessable) maybeAnchor;
-                return anchor.canBlockBeAccessed(toTp) || !isValidTarget(toTp, target, TravelSource.BLOCK);
+                if (!anchor.canBlockBeAccessed(toTp) && isValidTarget(toTp, target, TravelSource.BLOCK)) {
+                    return "not valid target";
+                }
+                return null;
             case STAFF:
             case STAFF_BLINK:
-                if (equippedItem == null || !(equippedItem.getItem() instanceof IItemOfTravel)) return false;
-                if (!((IItemOfTravel) equippedItem.getItem()).isActive(toTp, equippedItem)) return false;
+                if (equippedItem == null || !(equippedItem.getItem() instanceof IItemOfTravel)) return "not staff";
+                if (!((IItemOfTravel) equippedItem.getItem()).isActive(toTp, equippedItem)) return "staff not active";
                 int energy = ((IItemOfTravel) equippedItem.getItem()).canExtractInternal(equippedItem, powerUse);
-                return energy == -1 || energy == powerUse;
+                if (energy != -1 && energy != powerUse) {
+                    return "not enough power";
+                }
+                return null;
             case TELEPORT_STAFF_BLINK:
             case TELEPORT_STAFF:
                 // tp staff is creative version of traveling staff
                 // no energy check or anything else needed
                 // but the player must actually be equipped with one of these
-                return equippedItem != null && equippedItem.getItem() instanceof ItemTeleportStaff;
+                if (equippedItem != null && equippedItem.getItem() instanceof ItemTeleportStaff) {
```

Comment:
shouldn't the instanceof here be inverted if we are looking if the current equipped is a creative staff?

Reply:
fixed